src/GafferOSL/ShadingEngine.cpp

```diff
@@ -932,11 +932,16 @@ static const T *varyingValue( const IECore::CompoundData *points, const char *na
 
 } // namespace
 
-ShadingEngine::ShadingEngine( const IECoreScene::ShaderNetwork *shaderNetwork )
+ShadingEngine::ShadingEngine( const IECoreScene::ShaderNetwork *shaderNetwork, const CompoundObject *targetAttributes )
 	:	m_hash( shaderNetwork->Object::hash() ), m_timeNeeded( false ), m_unknownAttributesNeeded( false ), m_hasDeformation( false )
 {
 	ShaderNetworkPtr networkCopy = shaderNetwork->copy();
 	IECoreScene::ShaderNetworkAlgo::convertOSLComponentConnections( networkCopy.get(), OSL_VERSION );
+	if( targetAttributes )
+	{
+		networkCopy->applySubstitutions( targetAttributes );
+	}
```

Comment:
I'd assumed we'd be doing this in the cache getter in OSLShader.cpp, since we were trying to keep the substitution stuff outside of ShadingEngine.

Reply:
Fixed in 6b0990f ( using r-value reference constructor )